lapis2/src/main/kotlin/org/genspectrum/lapis/config/SequenceFilterFields.kt

```diff
@@ -0,0 +1,39 @@
+package org.genspectrum.lapis.config
+
+typealias FieldName = String
+
+data class SequenceFilterFields(val fields: Map<FieldName, SequenceFilterFieldType>) {
+    companion object {
+        private val nucleotideMutationsField = Pair("nucleotideMutations", SequenceFilterFieldType.MutationsList)
+
+        fun fromDatabaseConfig(databaseConfig: DatabaseConfig) = SequenceFilterFields(
+            fields = databaseConfig.schema.metadata
+                .map(::mapToSequenceFilterFields)
+                .flatten()
+                .toMap() + nucleotideMutationsField,
```

Comment:
This includes the implicit assumption that the config (more precisely `databaseConfig.schema.metadata` contains metadata columns for SILO. The computations here shows that "metadata columns in SILO" != "filters that LAPIS permits".
@chaoran-chen is that assumption justified?

Reply:
Yes, the `metadata` field in the config contains the metadata column names and types for SILO. For strings (and Pango lineages), the LAPIS filters will be named the same. For numbers and dates, there should be `<fieldname>From` and `<fieldname>To` filters.

(Does that answer your question?)